src/main/java/meteordevelopment/meteorclient/systems/modules/misc/Spam.java

```diff
@@ -114,7 +132,47 @@ private void onTick(TickEvent.Post event) {
                 i = messageI++;
             }
 
+
+            /* This var will be reassigned and outputted at the end
+             * The struct looks like this:
+             * /msg + playerNames[it] + text + bypass
+             */
             String text = messages.get().get(i);
+
+            // PMspam REALIZATION   ///////////////////////////////////////////////////////////////////////////////////////
+            // If our button is toggled         // TODO: List caching
+            if (PMspam.get()) {
+                if(mc.player != null) {
+                    // Var to store String[] of names
+                    String[] playerNames = new String[0];
+
+                    // if LOCAL SERVER
+                    if (mc.getServer() != null && mc.isConnectedToLocalServer()) {
+                        playerNames = mc.getServer().getPlayerNames();
+                    }
+                    // if MULTIPLAYER
+                    else if (mc.player.getEntityWorld().getPlayers() != null) {
+                        playerNames = mc.player.networkHandler.getCommandSource().getPlayerNames().toArray(new String[0]);
```

Comment:
Instead of calling `networkHandler.getCommandSource().getPlayerNames()`, you should be using `networkHandler.getPlayerList()`. `getPlayerNames()` allocates a new `List` every method call and internally uses `getPlayerList()`, so you can cut out the middle man and just use it directly. As a bonus, this returns an already cached list so you dont need to worry about it yourself.